common/achievement/index.ts

```diff
@@ -129,7 +129,10 @@ async function checkUserAchievement<ID extends ActionID>(userAchievement: UserAc
         const evaluationResultValue =
             typeof evaluationResult.resultObject[dataAggregationKey] === 'number' ? Number(evaluationResult.resultObject[dataAggregationKey]) : null;
         const awardedAchievement = await rewardUser(evaluationResultValue, userAchievement, event);
-        await createAchievement(awardedAchievement.template, userAchievement.userId, event.context);
+        const newAchievement = await createAchievement(awardedAchievement.template, userAchievement.userId, event.context);
+        if (newAchievement) {
+            await checkUserAchievement(newAchievement, event);
```

Comment:
When we create a new achievement we want to directly evaluate if it was already achieved, so we could generate the subsequent achievement.

This could only loop if we had endless acheivements in a group, which is not the case. the createAchievement function only works, if there are subsequent achievements in our currently checked achievement group.

Reply:
But can't the context change between these achievements? Every metric has it's own set of actions, which has it's own context.
This means that event A coming from action A will have a different context then event B coming from action B.

Reply:
You have a point here. We included a check for a sequential achievement being the last in the group to determine, whether to directly set its achievedAt value to the current Date or not.